downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -0,0 +1,59 @@
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+
+import java.util.*;
+
+public class CellularComponentAnnotationBuilder {
+
+    private static final List<String> speciesWithAlternateGOCompartment = new ArrayList<>(Arrays.asList("11676", "211044", "1491", "1392"));
+    private static final List<String> microbialSpeciesToExclude = new ArrayList<>(Arrays.asList("813", "562", "491", "90371", "1280", "5811"));
+    private static final String PROTEIN_BINDING_ANNOTATION = "0005515";
+
+    public static void processCellularComponents(GKInstance reactionInst) throws Exception {
+        Set<GKInstance> proteinInstances = GOAGeneratorUtilities.retrieveProteins(reactionInst);
+        processProteins(proteinInstances, reactionInst);
+    }
+
+    private static void processProteins(Set<GKInstance> proteinInstances, GKInstance reactionInst) throws Exception {
+        for (GKInstance proteinInst : proteinInstances) {
+            GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+            GKInstance speciesInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.species);
+            // Check if the protein has any disqualifying attributes.
+            boolean validProtein = EntityValidator.validateProtein(referenceEntityInst, speciesInst);
+            if (validProtein) {
+                String taxonIdentifier = ((GKInstance) speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+                // TODO: Rule check
+                if (!microbialSpeciesToExclude.contains(taxonIdentifier)) {
+
+                    // For CC annotations, alternateGOCompartments are disqualifying.
+                    //TODO: Rule check
+                    if (!speciesWithAlternateGOCompartment.contains(taxonIdentifier)) {
+                        // For CC, we are looking at ALL proteins in a ReactionlikeEvent
+                        generateGOCellularCompartmentLine(proteinInst, referenceEntityInst, reactionInst, taxonIdentifier);
+                    }
+                }
+            }
+        }
+    }
+
+    // This feeds into the GOA line generator. It formats that method call with attributes specific to CC annotations.
+    private static String generateGOCellularCompartmentLine(GKInstance proteinInst, GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+        String reactomeIdentifier = "REACTOME:" + ((GKInstance) reactionInst.getAttributeValue(ReactomeJavaConstants.stableIdentifier)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+        String goCellularCompartmentAccession = getCellularCompartmentGOAccession(proteinInst);
+        String goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, "C", goCellularCompartmentAccession, reactomeIdentifier, "TAS", taxonIdentifier);
```

Comment:
The annotation letters "C", "P", and "F", as well as the "TAS" and "EXP" evidence codes, could also be named constants

Reply:
Good point -- do you know what TAS and EXP stand for?

Reply:
LOL I guess you could just name the constant `EVIDENCE_CODE_TAS`, `EVIDENCE_CODE_EPX`, `EVIDENCE_CODE_C`, `EVIDENCE_CODE_F`, `EVIDENCE_CODE_P`, ... at least that way we'd know they are evidence codes.

Reply:
Haha, no need for that.  TAS stands for Traceable Author Statement.  EXP stands for Inferred From Experiment.

http://geneontology.org/docs/guide-go-evidence-codes/

Reply:
So... `String TRACEABLE_AUTHOR_STATEMENT = "TAS";` ?

Reply:
Yeah, that works.